add lightbox gallery and adjust site responsiveness for small screens

poncem91 opened the pull request:
Commit adds multiple thumbnail pictures where the one picture was before and opens up a lightbox gallery when a thumbnail is clicked.
Also makes small changes to site in general to make it responsive to smaller screens and avoid collapsing the navbar.

poncem91 committed:
add lightbox gallery and adjust site reponsiveness for small screens

Commit adds multiple thumbnail pictures where the one picture was before and opens up a lightbox gallery when a thumbnail is clicked.
Also makes small changes to site in general to make it responsive to smaller screens and avoid collapsing the navbar.

poncem91 committed:
add contact me section

Commit adds social media and site icons (generated by an external script) to the Contact Me section. These icons link to my online profiles.

peap commented on portfolio/src/main/webapp/index.html:
```diff
@@ -98,5 +144,63 @@ <h1>Work Experience</h1>
+                  <a class="interface-buttons" id="prev" onclick="changePic(-1)">&#10094;</a>
+                  <a class="interface-buttons" id="next" onclick="changePic(1)">&#10095;</a>
+                  <div class="lightbox-pic">
+                      <img src="/images/arlo1.jpg" alt="Arlo chilling in driveway">
```
You could also consider using `<figure>` and `<figcaption>` for these images: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/figcaption

poncem91 replied:
Thanks!!

peap commented on portfolio/src/main/webapp/script.js:
```diff
@@ -64,4 +64,49 @@ function scrollToSection(sectionId) {
+function showPic(picIndex) {
+  const picsArray = document.getElementsByClassName("lightbox-pic");
+
+  if (picIndex > picsArray.length) {
+    picIndex = 1;	
+  }
+  
+  else if (picIndex < 1) {
+  	picIndex = picsArray.length;
+  }
```
Consider using the remainder operator (`%`) to keep your index within bounds here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Arithmetic_Operators#Remainder

poncem91 replied:
This is a really clever way of wrapping around array index, thanks!!

peap commented on portfolio/src/main/webapp/index.html:
```diff
@@ -98,5 +144,63 @@ <h1>Work Experience</h1>
     </div>
+
+    <!-- Lightbox Gallery -->
+          <div id="lightbox">
```
Totally optional, but if you want a little extra work, try making it possible to close the lightbox with the escape key, in addition to the close link.

poncem91 replied:
Great idea! Added left and arrow key functionality as well while at it

peap reviewed (changes requested):
Looks great!

poncem91 committed:
Merge pull request #4 from poncem91/Mafe_ContactMeSection

add contact me section

poncem91 committed:
change lightbox implementation and use <figure> HTML tags

Commit uses the remainder operator to make sure the picture indexes stay within bounds in the lightbox implementation functions. Also makes use of the <figure> and <figcaption>.

poncem91 committed:
add keyboard functionality within lightbox gallery

Commit allows user to use left and right arrows in the keyboard and the escape key to use the lightbox gallery.

poncem91 committed:
Merge branch 'Mafe_LightboxGallery' of https://github.com/poncem91/step into Mafe_LightboxGallery

Merging local and remote branch

peap commented on portfolio/src/main/webapp/script.js:
```diff
@@ -81,32 +82,37 @@ function openLightbox(picIndex) {
 // Previous and Next picture functionality
 function changePic(byIndex) {
     currPic += byIndex;
+    currPic = (currPic + picsArray.length) % picsArray.length;
```
Adding `picsArray.length` to `currPic` is redundant when taking the remainder of `picsArray.length`. (It's like multiplying by one or adding zero.) So you can just say `currPic %= picsArray.length`.

poncem91 replied:
I tried that at first, but when trying to chagePic from index 0 to 11, it computed -1 % 12 and outputted -1. I think % acts differently in JS, but let me know if I'm doing something wrong or if there's a better way to do this

peap replied:
Ah, yes, my bad. What you had originally is good; adding the length before taking the remainder is the way to go.
